scripts/kubestellar-snapshot.sh

```diff
@@ -0,0 +1,567 @@
+#!/usr/bin/env bash
+# Copyright 2024 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Exit on error
+set -e
+
+
+# Global variables
+timestamp="$(date +%F_%T)"
+output_folder="/tmp/kubestellar-snapshot"
+
+
+# Script info
+SCRIPT_NAME="KubeStellar Snapshot"
+SCRIPT_VERSION="0.10"
+
+
+# Colors
+COLOR_NONE="\033[0m"
+COLOR_RED="\033[1;31m"
+COLOR_GREEN="\033[1;32m"
+COLOR_BLUE="\033[94m"
+COLOR_YELLOW="\033[1;33m"
+COLOR_PURPLE="\033[1;35m"
+COLOR_ERROR="${COLOR_RED}"
+COLOR_STATUS_TRUE="${COLOR_GREEN}"
+COLOR_STATUS_FALSE="${COLOR_RED}"
+COLOR_INFO="${COLOR_BLUE}"
+COLOR_TITLE="${COLOR_YELLOW}"
+COLOR_YAML="${COLOR_PURPLE}"
+
+
+# Command line arguments
+arg_kubeconfig=""
+arg_context=""
+arg_logs=false
+arg_yaml=false
+arg_verbose=false
+
+
+# Display the command line help
+display_help() {
+  cat << EOF
+Usage: $0 [--kubeconfig|-K <filename>] [--context|-C <name>] [--logs|-L] [--yaml|-Y] [--verbose|-V] [--version|-v] [--help|-h] [-X]
```

Comment:
As long as you are showing a listing of the options flags with descriptions, you do not need to repeat the flags in the synopsis. You could follow the pattern seen in the example below.

```console
mspreitz@mjs13 kubestellar % make -h
Usage: make [options] [target] ...
Options:
  -b, -m                      Ignored for compatibility.
  -B, --always-make           Unconditionally make all targets.
  -C DIRECTORY, --directory=DIRECTORY
                              Change to DIRECTORY before doing anything.
  -d                          Print lots of debugging information.
...
```

Reply:
ok, I have changed it to the proposed way to simplify maintenance